swiftwinrt/Resources/Support/IWeakReferenceSource.swift

```diff
@@ -0,0 +1,68 @@
+import C_BINDINGS_MODULE
+
+internal let IID_IWeakReferenceSource: IID = IID(Data1: 0x00000038, Data2: 0x0000, Data3: 0x0000, Data4: (0xC0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46)) // 00000038-0000-0000-C000-000000000046
+
+fileprivate extension IUnknownRef {
+    func copyTo(_ riid: REFIID?, _ ppvObj: UnsafeMutablePointer<LPVOID?>?) -> HRESULT {
+        self.borrow.pointee.lpVtbl.pointee.QueryInterface(self.borrow, riid, ppvObj)
+    }
+}
+
+func makeWeakReferenceSource(_ object: AnyObject, _ result: UnsafeMutablePointer<UnsafeMutableRawPointer?>) throws {
+    let wrapper = WeakReferenceSourceWrapper(object)
+    try wrapper.toABI {
+        _ = $0.pointee.lpVtbl.pointee.AddRef($0)
+        result.pointee = UnsafeMutableRawPointer($0)
+    }
+}
+
+fileprivate class WeakReferenceSourceWrapper: WinRTAbiBridgeWrapper<IWeakReferenceSourceBridge> {
+    init(_ object: AnyObject){
+        super.init(IWeakReferenceSourceBridge.makeAbi(), object)
+    }
+}
+
+fileprivate enum IWeakReferenceSourceBridge: AbiBridge {
+    typealias CABI = C_IWeakReferenceSource
+    typealias SwiftProjection = AnyObject
+
+    static func makeAbi() -> C_IWeakReferenceSource {
+        return C_IWeakReferenceSource(lpVtbl: &IWeakReferenceSourceVTable)
+    }
+
+    static func from(abi: ComPtr<C_IWeakReferenceSource>?) -> AnyObject? {
+        fatalError("Not needed")
+    }
+}
+
+fileprivate var IWeakReferenceSourceVTable: C_IWeakReferenceSourceVtbl = .init(
+    QueryInterface: { pUnk, riid, ppvObject in
+        guard let pUnk, let riid, let ppvObject else { return E_INVALIDARG }
+        switch riid.pointee {
+        case IID_IWeakReferenceSource:
+            _ = pUnk.pointee.lpVtbl.pointee.AddRef(pUnk)
+            return S_OK
+        default:
+            guard let obj = WeakReferenceSourceWrapper.tryUnwrapFromBase(raw: pUnk) else { return E_NOINTERFACE }
+            fatalError("\(#function): Not implemented: something like obj.QueryInterface")
```

Comment:
@stevenbrix , I'm blocked here. I think I need to have an `-Impl` object that keeps a COMRef to the object somehow?

Reply:
No but I could piggyback on the `AnyWrapper.queryInterface` implementation.